packages/nimbus/src/components/data-table/components/data-table.manager.tsx

```diff
@@ -0,0 +1,240 @@
+
+  const handleSettingsChange = (
+    action: (typeof UPDATE_ACTIONS)[keyof typeof UPDATE_ACTIONS] | undefined
+  ) => {
+    if (onSettingsChange) {
+      onSettingsChange(action);
+    }
+  };
+
+  const hiddenItems = useMemo(() => {
+    return columns
+      .filter((item) => !visibleColumns?.includes(item.id))
+      .map((item) => ({
+        key: item.id,
+        id: item.id,
+        label: item.header,
+      }));
+  }, [visibleColumns, columns]);
+
+  // Create a map of column IDs to preserve order from visibleColumns
+  const visibleItems = useMemo(() => {
+    if (!visibleColumns || visibleColumns.length === 0) {
+      return [];
+    }
+
+    const columnMap = new Map(columns.map((col) => [col.id, col]));
+
+    // Map visible column IDs to their full column data, preserving order
+    return visibleColumns
+      .map((id) => {
+        const col = columnMap.get(id);
+        return col
+          ? {
+              key: col.id,
+              id: col.id,
+              label: col.header,
+            }
+          : null;
+      })
+      .filter((item): item is NonNullable<typeof item> => item !== null);
+  }, [columns, visibleColumns]);
+
+  const handleResetColumns = useCallback(() => {
+    // Reset to original column order and visibility
+    if (onColumnsChange) {
+      const initialVisibleCols = initialVisibleColumnsRef.current;
+
+      // If there were initially visible columns specified, restore them
+      if (initialVisibleCols && initialVisibleCols.length > 0) {
+        const columnMap = new Map(
+          initialColumnsRef.current.map((col) => [col.id, col])
+        );
+
+        // Map the initial visible column IDs back to their full column objects
+        const restoredColumns = initialVisibleCols
+          .map((id) => columnMap.get(id))
+          .filter(
+            (col): col is DataTableColumnItem<Record<string, unknown>> =>
+              col !== undefined
+          );
+
+        // Update the refs
+        lastNotifiedColumnsRef.current = initialVisibleCols;
+
+        // Notify parent
+        onColumnsChange(restoredColumns);
+      } else {
+        // No initial visibleColumns was set, so restore all columns
+        lastNotifiedColumnsRef.current = initialColumnsRef.current.map(
+          (col) => col.id
+        );
+        onColumnsChange(initialColumnsRef.current);
+      }
+    }
+  }, [onColumnsChange]);
+
+  const defaultTrigger = (
+    <Tooltip.Root>
+      <Tooltip.Content placement="top">
+        {formatMessage(messages.settings)}
+      </Tooltip.Content>
+      <IconButton
+        variant="ghost"
+        tone="primary"
+        size="xs"
+        aria-label={formatMessage(messages.settings)}
+        onClick={() => setIsOpen(true)}
+      >
+        <Settings />
+      </IconButton>
+    </Tooltip.Root>
+  );
+
+  return (
+    <>
+      {defaultTrigger}
+      <Drawer.Root isOpen={isOpen} onOpenChange={setIsOpen} placement="right">
+        <Drawer.Content width="640px">
```

Comment:
nit: can we use a design token for the width here? either `xl` (574px) or `2xl` (672px)?